src/AbstractSQLOptimiser.ts

```diff
@@ -672,6 +677,24 @@ const typeRules: Dictionary<MatchFn> = {
 	ToDate: matchArgs('ToDate', DateValue),
 	DateTrunc: matchArgs('DateTrunc', TextValue, DateValue),
 	ToTime: matchArgs('ToTime', DateValue),
+	ExtractJSONPathAsText: (args) => {
+		checkMinArgs('ExtractJSONPathAsText', args, 1);
+		const column = TextValue(getAbstractSqlQuery(args, 0));
+		const path = args.slice(1).map((arg) => {
+			if (!isAbstractSqlQuery(arg)) {
+				throw new SyntaxError(
+					`Expected AbstractSqlQuery array but got ${typeof arg}`,
+				);
+			}
+			return ArrayValue(arg);
+		});
+		return ['ExtractJSONPathAsText', column, ...path];
+	},
+	TextArray: (args) => {
+		// Allow for populated and empty arrays
+		return ['TextArray', ...args];
```

Comment:
To enforce each element of the array is a text value:
```suggestion
		return ['TextArray', ...args.map(TextValue)];
```